src/dotnet/commands/dotnet-migrate/MigrateCommand.cs

```diff
@@ -46,7 +47,7 @@ public int Execute()
                 Console.WriteLine($"Migrating project {project}..");
                 var projectDirectory = Path.GetDirectoryName(project);
                 var outputDirectory = projectDirectory;
-                var migrationSettings = new MigrationSettings(projectDirectory, outputDirectory, sdkVersion, msBuildTemplate, _xprojFilePath);
+                var migrationSettings = new MigrationSettings(projectDirectory, outputDirectory, sdkVersion, msBuildTemplate.DeepClone(), _xprojFilePath);
```

Comment:
Fixed. Moved the `DeepClone` to `MigrationSettings` but this does not completely remove the burden from the caller. The caller still needs to make sure that they pass the correct msbuild template.